spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -538,68 +543,127 @@ protected WriterFactory(Broadcast<Table> tableBroadcast, FileFormat format, long
     @Override
     public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long epochId) {
       Table table = tableBroadcast.value();
-
-      OutputFileFactory fileFactory = OutputFileFactory.builderFor(table, partitionId, taskId).format(format).build();
-      SparkAppenderFactory appenderFactory = SparkAppenderFactory.builderFor(table, writeSchema, dsSchema).build();
-
       PartitionSpec spec = table.spec();
       FileIO io = table.io();
 
+      OutputFileFactory fileFactory = OutputFileFactory.builderFor(table, partitionId, taskId)
+          .format(format)
+          .build();
+      SparkFileWriterFactory writerFactory = SparkFileWriterFactory.builderFor(table)
```

Comment:
Does this need to be created each time we create a writer? It might not matter if we only create one per task since there's nothing to share?

Reply:
I replicated the old behavior with `SparkAppenderFactory` but I can check whether it is something we can create per task.